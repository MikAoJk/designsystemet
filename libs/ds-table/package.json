{
  "name": "@skatteetaten/ds-table",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.5.2",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^13",
    "@skatteetaten/ds-core-utils": "*",
    "@skatteetaten/ds-icons": "*",
    "@skatteetaten/ds-buttons": "*"
  },
  "scripts": {}
}
