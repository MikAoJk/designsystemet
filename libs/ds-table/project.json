{
  "name": "ds-table",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-table/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.{ts,tsx,js,jsx}",
          "libs/ds-table/package.json"
        ]
      }
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-table",
        "tsConfig": "libs/ds-table/tsconfig.lib.json",
        "project": "libs/ds-table/package.json",
        "entryFile": "libs/ds-table/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-table/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-table/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-table/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.css",
          "libs/ds-table/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-table"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-table {args.switches}",
        "cwd": ""
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
