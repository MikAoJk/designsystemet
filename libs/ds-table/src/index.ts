export * from './Table/Table';
export * from './Table/Table.types';
