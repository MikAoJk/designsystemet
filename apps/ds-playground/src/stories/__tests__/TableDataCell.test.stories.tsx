import { Table } from '@skatteetaten/ds-table';
import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { within } from '@storybook/testing-library';

export default {
  component: Table.DataCell,
  title: 'Tester/Table/DataCell',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    alignment: { table: { disable: true } },
    as: { table: { disable: true } },
    colSpan: { table: { disable: true } },
    rowSpan: { table: { disable: true } },
    scope: { table: { disable: true } },
    children: { table: { disable: true } },
  },
} as ComponentMeta<typeof Table.DataCell>;

const defaultArgs = {
  children: 'cake',
};

const Template: ComponentStory<typeof Table.DataCell> = (args) => (
  <div data-test-block>
    {/* eslint-disable-next-line testing-library/no-node-access */}
    <Table.DataCell {...args}>{args.children}</Table.DataCell>
  </div>
);

// Når Table har en ref, så får dom table elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLTableCellElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const table = canvas.getByRole('cell');
  await expect(table).toBeInTheDocument();
  await expect(table).toHaveAttribute('id', 'dummyIdForwardedFromRef');
};

// Når TableDataCell har en id, så har th-element id
// Når TableDataCell har en custom CSS, så vises custom stil
// Når TableDataCell har en lang, så har th-element lang
// Når TableDataCell har dataTestid, så har th-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes (FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.parameters = {
  imageSnapshot: { disable: true },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const dataCell = canvas.getByRole('cell');
  await expect(dataCell).toHaveClass('dummyClassname');
  await expect(dataCell).toHaveAttribute('id', 'htmlId');
  await expect(dataCell).toHaveAttribute('lang', 'nb');
  await expect(dataCell).toHaveAttribute('data-testid', '123ID');
};

// Når TableDataCell har scope, så har th-elementet attributet satt
// Når TableDataCell har as, så render den som forventet element
export const WithThAndScope = Template.bind({});
WithThAndScope.storyName = 'With Scope And As th (TableRow B2, TableHeader B1)';
WithThAndScope.args = {
  ...defaultArgs,
  as: 'th',
  scope: 'row',
};
WithThAndScope.argTypes = {
  ...WithThAndScope.argTypes,
  as: { table: { disable: false } },
  scope: { table: { disable: false } },
};
WithThAndScope.parameters = {
  imageSnapshot: { disable: true },
};
WithThAndScope.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const dataCell = canvas.getByRole('rowheader');
  await expect(dataCell).toHaveAttribute('scope', 'row');
};

// Når TableDataCell har children, så render den som forventet
export const WithChildren = Template.bind({});
WithChildren.storyName = 'With children (TableRow A1)';
WithChildren.args = {
  children: 'cellChildren',
};
WithChildren.argTypes = {
  children: { table: { disable: false } },
};

WithChildren.parameters = {
  imageSnapshot: { disable: true },
};
WithChildren.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  await expect(canvas.getByText('cellChildren')).toBeInTheDocument();
};
