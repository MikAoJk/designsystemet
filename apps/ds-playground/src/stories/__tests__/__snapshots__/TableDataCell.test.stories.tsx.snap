// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Table/DataCell WithAttributes play-test 1`] = `
<div data-test-block="true">
  <td id="htmlId"
      class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  dummyClassname"
      lang="nb"
      data-testid="123ID"
  >
    cake
  </td>
</div>
`;

exports[`Tester/Table/DataCell WithChildren play-test 1`] = `
<div data-test-block="true">
  <td class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  ">
    cellChildren
  </td>
</div>
`;

exports[`Tester/Table/DataCell WithRef play-test 1`] = `
<div data-test-block="true">
  <td class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  "
      id="dummyIdForwardedFromRef"
  >
    cake
  </td>
</div>
`;

exports[`Tester/Table/DataCell WithThAndScope play-test 1`] = `
<div data-test-block="true">
  <th class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  "
      scope="row"
  >
    cake
  </th>
</div>
`;
